pytests/test_recovery.py

```diff
@@ -1,10 +1,9 @@
 from threading import Event
 
-from bytewax import AdvanceTo, Dataflow, Emit, run_main
-from bytewax.recovery import SqliteRecoveryConfig
-
 from pytest import fixture, raises
 
+from bytewax import AdvanceTo, Dataflow, Emit, run_main
+from bytewax.recovery import SqliteRecoveryConfig
 
 RECOVERY_CONFIG_TYPES = [
     "SqliteRecoveryConfig",
```

Comment:
Should we add the KafkaRecoveryConfig?

Reply:
Yes, we should, but it would be hard because we'd have to encode some way to startup a Kafka cluster for these tests to connect to. SQLite works here because everything is self contained within the connecting process, and I was thinking we might have others that fit that shape (like `DumpToFlatFileRecoveryConfig`). At some point in the future we could have integration tests that include these kind of not-just-Python testing dependencies. Redpanda might be useful here as it's a little less "sprawling" than a Java-based Kafka install. It's also possible there's some pure-Python Kafka-protocol compatible mock we could use, but I didn't find one. Also in general we're going to have a harder time doing mock-based testing because our recovery code is in Rust, so we can't use the standard Python mocking tools to replace the inner calls to some "testing" datastore, and our tests will need to use something that supports the actual connectivity mechanism.